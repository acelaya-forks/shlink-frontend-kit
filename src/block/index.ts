export * from './Message';
export * from './Result';
export * from './SimpleCard';
