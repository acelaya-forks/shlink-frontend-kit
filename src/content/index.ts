export * from './Details';
export * from './Listbox';
export * from './Table';
