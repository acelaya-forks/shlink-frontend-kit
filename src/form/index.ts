export * from './Checkbox';
export * from './ToggleSwitch';
export * from './InputFormGroup';
export * from './LabeledFormGroup';
export * from './SearchField';
