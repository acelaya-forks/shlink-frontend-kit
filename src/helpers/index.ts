export * from './numbers';
export * from './pagination';
