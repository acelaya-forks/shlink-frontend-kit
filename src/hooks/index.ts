import { useRef, useState } from 'react';
import { v4 as uuid } from 'uuid';

type ToggleResult = [boolean, () => void, () => void, () => void];

export const useToggle = (initialValue = false): ToggleResult => {
  const [flag, setFlag] = useState<boolean>(initialValue);
  return [flag, () => setFlag(!flag), () => setFlag(true), () => setFlag(false)];
};

export const useDomId = (): string => {
  const { current: id } = useRef(`dom-${uuid()}`);
  return id;
};

export const useElementRef = <T>() => useRef<T | null>(null);
