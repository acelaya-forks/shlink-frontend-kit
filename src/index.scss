@import './utils/ResponsiveTable';
@import './theme/theme';
