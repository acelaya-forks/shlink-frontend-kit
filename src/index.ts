export * from './block';
export * from './form';
export * from './hooks';
export * from './navigation';
export * from './ordering';
