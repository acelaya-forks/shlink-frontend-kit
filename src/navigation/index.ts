export * from './Dropdown';
export * from './LinkButton';
export * from './Menu';
export * from './NavBar';
export * from './NavPills';
export * from './Paginator';
export * from './RowDropdown';
