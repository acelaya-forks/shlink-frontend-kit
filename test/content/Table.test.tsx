import { render } from '@testing-library/react';
import type { TableProps } from '../../src';
import { Table } from '../../src';
import { checkAccessibility } from '../__helpers__/accessibility';

describe('<Table />', () => {
  const setUp = (props: Omit<Partial<TableProps>, 'children'> = {}) => render(
    <Table
      header={
        <Table.Row>
          <Table.Cell>Foo</Table.Cell>
          <Table.Cell>Bar</Table.Cell>
          <Table.Cell>Baz</Table.Cell>
        </Table.Row>
      }
      {...props}
    >
      <Table.Row>
        <Table.Cell columnName="Foo">Foo</Table.Cell>
        <Table.Cell columnName="Bar">Bar</Table.Cell>
        <Table.Cell columnName="Baz">Baz</Table.Cell>
      </Table.Row>
      <Table.Row>
        <Table.Cell columnName="Foo">Foo</Table.Cell>
        <Table.Cell columnName="Bar">Bar</Table.Cell>
        <Table.Cell columnName="Baz">Baz</Table.Cell>
      </Table.Row>
      <Table.Row>
        <Table.Cell>Foo</Table.Cell>
        <Table.Cell>Bar</Table.Cell>
        <Table.Cell>Baz</Table.Cell>
      </Table.Row>
    </Table>,
  );

  it('passes a11y checks', () => checkAccessibility(setUp()));

  it.each([
    { responsive: false },
    {
      footer: (
        <Table.Row>
          <Table.Cell type="td">Foo footer</Table.Cell>
          <Table.Cell type="td">Bar footer</Table.Cell>
          <Table.Cell type="td">Baz footer</Table.Cell>
        </Table.Row>
      ),
    },
    { size: 'sm' as const },
    { size: 'lg' as const },
  ])('renders as expected based on provided props', (props) => {
    const { container } = setUp(props);
    expect(container).toMatchSnapshot();
  });
});
