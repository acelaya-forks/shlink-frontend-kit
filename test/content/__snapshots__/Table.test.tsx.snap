// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Table /> > renders as expected based on provided props 1`] = `
<div>
  <table
    class="tw:w-full"
  >
    <thead
      class=""
    >
      <tr
        class="tw:group"
      >
        <th
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:border-b-1 tw:text-left"
        >
          Foo
        </th>
        <th
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:border-b-1 tw:text-left"
        >
          Bar
        </th>
        <th
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:border-b-1 tw:text-left"
        >
          Baz
        </th>
      </tr>
    </thead>
    <tbody
      class=""
    >
      <tr
        class="tw:group tw:hover:bg-lm-primary tw:dark:hover:bg-dm-primary tw:group-[&]/card:hover:bg-lm-secondary tw:dark:group-[&]/card:hover:bg-dm-secondary"
      >
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:border-b-1"
        >
          Foo
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:border-b-1"
        >
          Bar
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:border-b-1"
        >
          Baz
        </td>
      </tr>
      <tr
        class="tw:group tw:hover:bg-lm-primary tw:dark:hover:bg-dm-primary tw:group-[&]/card:hover:bg-lm-secondary tw:dark:group-[&]/card:hover:bg-dm-secondary"
      >
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:border-b-1"
        >
          Foo
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:border-b-1"
        >
          Bar
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:border-b-1"
        >
          Baz
        </td>
      </tr>
      <tr
        class="tw:group tw:hover:bg-lm-primary tw:dark:hover:bg-dm-primary tw:group-[&]/card:hover:bg-lm-secondary tw:dark:group-[&]/card:hover:bg-dm-secondary"
      >
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:border-b-1"
        >
          Foo
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:border-b-1"
        >
          Bar
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:border-b-1"
        >
          Baz
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<Table /> > renders as expected based on provided props 2`] = `
<div>
  <table
    class="tw:w-full"
  >
    <thead
      class="tw:hidden tw:lg:table-header-group"
    >
      <tr
        class="tw:group tw:lg:table-row tw:flex tw:flex-col tw:lg:border-0 tw:border-y-2 tw:border-lm-border tw:dark:border-dm-border"
      >
        <th
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:text-left"
        >
          Foo
        </th>
        <th
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:text-left"
        >
          Bar
        </th>
        <th
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:text-left"
        >
          Baz
        </th>
      </tr>
    </thead>
    <tbody
      class="tw:lg:table-row-group tw:flex tw:flex-col tw:gap-y-3"
    >
      <tr
        class="tw:group tw:lg:table-row tw:flex tw:flex-col tw:lg:border-0 tw:border-y-2 tw:border-lm-border tw:dark:border-dm-border tw:hover:bg-lm-primary tw:dark:hover:bg-dm-primary tw:group-[&]/card:hover:bg-lm-secondary tw:dark:group-[&]/card:hover:bg-dm-secondary"
      >
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
          data-column="Foo"
        >
          Foo
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
          data-column="Bar"
        >
          Bar
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
          data-column="Baz"
        >
          Baz
        </td>
      </tr>
      <tr
        class="tw:group tw:lg:table-row tw:flex tw:flex-col tw:lg:border-0 tw:border-y-2 tw:border-lm-border tw:dark:border-dm-border tw:hover:bg-lm-primary tw:dark:hover:bg-dm-primary tw:group-[&]/card:hover:bg-lm-secondary tw:dark:group-[&]/card:hover:bg-dm-secondary"
      >
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
          data-column="Foo"
        >
          Foo
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
          data-column="Bar"
        >
          Bar
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
          data-column="Baz"
        >
          Baz
        </td>
      </tr>
      <tr
        class="tw:group tw:lg:table-row tw:flex tw:flex-col tw:lg:border-0 tw:border-y-2 tw:border-lm-border tw:dark:border-dm-border tw:hover:bg-lm-primary tw:dark:hover:bg-dm-primary tw:group-[&]/card:hover:bg-lm-secondary tw:dark:group-[&]/card:hover:bg-dm-secondary"
      >
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
        >
          Foo
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
        >
          Bar
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
        >
          Baz
        </td>
      </tr>
    </tbody>
    <tfoot
      class="tw:lg:table-row-group tw:flex tw:flex-col tw:gap-y-3 tw:mt-4"
    >
      <tr
        class="tw:group tw:lg:table-row tw:flex tw:flex-col tw:lg:border-0 tw:border-y-2 tw:border-lm-border tw:dark:border-dm-border"
      >
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
        >
          Foo footer
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
        >
          Bar footer
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-2 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
        >
          Baz footer
        </td>
      </tr>
    </tfoot>
  </table>
</div>
`;

exports[`<Table /> > renders as expected based on provided props 3`] = `
<div>
  <table
    class="tw:w-full"
  >
    <thead
      class="tw:hidden tw:lg:table-header-group"
    >
      <tr
        class="tw:group tw:lg:table-row tw:flex tw:flex-col tw:lg:border-0 tw:border-y-2 tw:border-lm-border tw:dark:border-dm-border"
      >
        <th
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-1 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:text-left"
        >
          Foo
        </th>
        <th
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-1 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:text-left"
        >
          Bar
        </th>
        <th
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-1 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:text-left"
        >
          Baz
        </th>
      </tr>
    </thead>
    <tbody
      class="tw:lg:table-row-group tw:flex tw:flex-col tw:gap-y-3"
    >
      <tr
        class="tw:group tw:lg:table-row tw:flex tw:flex-col tw:lg:border-0 tw:border-y-2 tw:border-lm-border tw:dark:border-dm-border tw:hover:bg-lm-primary tw:dark:hover:bg-dm-primary tw:group-[&]/card:hover:bg-lm-secondary tw:dark:group-[&]/card:hover:bg-dm-secondary"
      >
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-1 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
          data-column="Foo"
        >
          Foo
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-1 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
          data-column="Bar"
        >
          Bar
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-1 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
          data-column="Baz"
        >
          Baz
        </td>
      </tr>
      <tr
        class="tw:group tw:lg:table-row tw:flex tw:flex-col tw:lg:border-0 tw:border-y-2 tw:border-lm-border tw:dark:border-dm-border tw:hover:bg-lm-primary tw:dark:hover:bg-dm-primary tw:group-[&]/card:hover:bg-lm-secondary tw:dark:group-[&]/card:hover:bg-dm-secondary"
      >
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-1 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
          data-column="Foo"
        >
          Foo
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-1 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
          data-column="Bar"
        >
          Bar
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-1 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
          data-column="Baz"
        >
          Baz
        </td>
      </tr>
      <tr
        class="tw:group tw:lg:table-row tw:flex tw:flex-col tw:lg:border-0 tw:border-y-2 tw:border-lm-border tw:dark:border-dm-border tw:hover:bg-lm-primary tw:dark:hover:bg-dm-primary tw:group-[&]/card:hover:bg-lm-secondary tw:dark:group-[&]/card:hover:bg-dm-secondary"
      >
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-1 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
        >
          Foo
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-1 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
        >
          Bar
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-1 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
        >
          Baz
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<Table /> > renders as expected based on provided props 4`] = `
<div>
  <table
    class="tw:w-full"
  >
    <thead
      class="tw:hidden tw:lg:table-header-group"
    >
      <tr
        class="tw:group tw:lg:table-row tw:flex tw:flex-col tw:lg:border-0 tw:border-y-2 tw:border-lm-border tw:dark:border-dm-border"
      >
        <th
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-3 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:text-left"
        >
          Foo
        </th>
        <th
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-3 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:text-left"
        >
          Bar
        </th>
        <th
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-3 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:text-left"
        >
          Baz
        </th>
      </tr>
    </thead>
    <tbody
      class="tw:lg:table-row-group tw:flex tw:flex-col tw:gap-y-3"
    >
      <tr
        class="tw:group tw:lg:table-row tw:flex tw:flex-col tw:lg:border-0 tw:border-y-2 tw:border-lm-border tw:dark:border-dm-border tw:hover:bg-lm-primary tw:dark:hover:bg-dm-primary tw:group-[&]/card:hover:bg-lm-secondary tw:dark:group-[&]/card:hover:bg-dm-secondary"
      >
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-3 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
          data-column="Foo"
        >
          Foo
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-3 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
          data-column="Bar"
        >
          Bar
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-3 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
          data-column="Baz"
        >
          Baz
        </td>
      </tr>
      <tr
        class="tw:group tw:lg:table-row tw:flex tw:flex-col tw:lg:border-0 tw:border-y-2 tw:border-lm-border tw:dark:border-dm-border tw:hover:bg-lm-primary tw:dark:hover:bg-dm-primary tw:group-[&]/card:hover:bg-lm-secondary tw:dark:group-[&]/card:hover:bg-dm-secondary"
      >
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-3 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
          data-column="Foo"
        >
          Foo
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-3 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
          data-column="Bar"
        >
          Bar
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-3 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
          data-column="Baz"
        >
          Baz
        </td>
      </tr>
      <tr
        class="tw:group tw:lg:table-row tw:flex tw:flex-col tw:lg:border-0 tw:border-y-2 tw:border-lm-border tw:dark:border-dm-border tw:hover:bg-lm-primary tw:dark:hover:bg-dm-primary tw:group-[&]/card:hover:bg-lm-secondary tw:dark:group-[&]/card:hover:bg-dm-secondary"
      >
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-3 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
        >
          Foo
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-3 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
        >
          Bar
        </td>
        <td
          class="tw:border-lm-border tw:dark:border-dm-border tw:p-3 tw:block tw:lg:table-cell tw:not-last:border-b-1 tw:lg:border-b-1 tw:before:lg:hidden tw:before:content-[attr(data-column)] tw:before:font-bold tw:before:mr-1"
        >
          Baz
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
